### PR TITLE
MINIFI-388 symlink versioned libcrypto/libssl to non-versioned path s…

### PR DESCRIPTION
**Pull request by achristianson**
…o that civet can find them when TLS is used

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -100,6 +100,10 @@ RUN mkdir -p $MINIFI_BASE_DIR
 COPY --from=builder ${MINIFI_HOME} ${MINIFI_HOME}
 RUN chown -R ${USER}:${USER} /opt/minifi
 
+# Link crypto libs to known locations for civet
+RUN ln -s /usr/lib/libcrypto.so.1.0.0 /usr/lib/libcrypto.so
```
Comment (phrocker) [inline docker/Dockerfile]: why not set the make option with CIVET? It seems like a bandaid that could come apart if something within the container changed. Could it be such that our make CRYPTO_LIB was determined via an embedded command?

Comment (achristianson) [inline docker/Dockerfile]: Looking into it... that would be a better fix if it works out.

**Comment by achristianson** [comment]
Updated to disable dynamic loading of TLS libs in civet, install libressl instead of openssl to resolve conflict with libcurl deps, and backport fix for civetweb which fixes compatibility with libressl.

Basically, libcurl and civet were using conflicting TLS versions. This worked due to dynamic loading of openssl libs, but turning that off and having both depend on libressl removes the duplication while also solving MINIFI-388.

This should be more of a "proper"/lasting fix. The civet patch backport will go away on the next version of civet.

**Comment by achristianson** [comment]
Looks like we have an issue now with civet/TLS in travis. Looking into it.

**Comment by achristianson** [comment]
Issue was addressed by an upstream commit in civetweb: https://github.com/civetweb/civetweb/commit/e849ce4b54c09d5b4441e371f17cf13368ac2234. Backported that commit which should fix the travis/ubuntu trusty issue.

File: thirdparty/civetweb-1.9.1/src/civetweb.c [hunk 2]
```diff
@@ -11856,7 +11859,24 @@ ssl_get_client_cert_info(struct mg_connection *conn)
 
 		/* Calculate SHA1 fingerprint and store as a hex string */
 		ulen = 0;
-		ASN1_digest((int (*)())i2d_X509, digest, (char *)cert, buf, &ulen);
+
+		/* ASN1_digest is deprecated. Do the calculation manually,
```
Comment (phrocker) [inline thirdparty/civetweb-1.9.1/src/civetweb.c]: is this also a backport?

Comment (achristianson) [inline thirdparty/civetweb-1.9.1/src/civetweb.c]: Yes.

Comment (achristianson) [inline thirdparty/civetweb-1.9.1/src/civetweb.c]: https://github.com/civetweb/civetweb/commit/fb9ef36bb295cf489aecd79c8bb9c6eac11e896e is where it comes from.

Comment (phrocker) [reply thirdparty/civetweb-1.9.1/src/civetweb.c]: Sorry I missed this response. I was concerned about using custom code in the third party dir, but with that knowledge, with that knowledge it's a non issue. I'll +1 and merge, thanks!

**Comment by calebj** [comment]
This doesn't build for me without backporting the `#define` for EVP_Digest and the matching entry in `crypto_sw`. I added these in 66dc8f039ca0fa1af362c11a5bcc43475456bc74; should I create another PR?

**Comment by phrocker** [comment]
@calebj sure! Can you tell me what distro you are building on?

**Comment by calebj** [comment]
I'm building on Cloud9, which has Ubuntu 14.04 with GCC 4.8.4, against OpenSSL version is 1.0.1f-1ubuntu2.22 if it helps.

The specific error was:
```
[ 45%] Building C object thirdparty/civetweb-1.9.1/src/CMakeFiles/c-library.dir/civetweb.c.o
/home/ubuntu/workspace/thirdparty/civetweb-1.9.1/src/civetweb.c: In function ‘ssl_get_client_cert_info’:
/home/ubuntu/workspace/thirdparty/civetweb-1.9.1/src/civetweb.c:11873:4: error: implicit declaration of function ‘EVP_Digest’ [-Wimplicit-function-declaration]
    if (!EVP_Digest(
    ^
make[2]: *** [thirdparty/civetweb-1.9.1/src/CMakeFiles/c-library.dir/civetweb.c.o] Error 1
```

**Comment by phrocker** [comment]
@calebj Thanks. I know some of us build with Ubuntu 14 and have run into similar issues. If you submit a PR I'll spin up an instance, verify, and can merge. Thanks!

**Pull request closed without merging**